src/nvenc/nvenc_base.cpp

```diff
@@ -293,6 +298,30 @@ namespace nvenc {
         fill_vui(format_config.hevcVUIParameters);
         break;
       }
+
+      case 2: {
+        // AV1
+        auto &format_config = enc_config.encodeCodecConfig.av1Config;
+        format_config.repeatSeqHdr = 1;
+        format_config.idrPeriod = NVENC_INFINITE_GOPLENGTH;
+        format_config.chromaFormatIDC = 1;  // YUV444 not supported by NVENC yet
+        if (buffer_is_10bit()) {
+          format_config.inputPixelBitDepthMinus8 = 2;
+          format_config.pixelBitDepthMinus8 = 2;
+        }
+        format_config.colorPrimaries = colorspace.primaries;
+        format_config.transferCharacteristics = colorspace.tranfer_function;
+        format_config.matrixCoefficients = colorspace.matrix;
+        format_config.colorRange = colorspace.full_range;
+        set_ref_frames(format_config.maxNumRefFramesInDPB, format_config.numBwdRefs, 8);
+        set_minqp_if_enabled(config.min_qp_av1);
+
+        if (client_config.slicesPerFrame > 1) {
+          // FIXME: Not supported yet
+          BOOST_LOG(warning) << "Client-controlled slicing is not yet supported by NVENC for AV1";

```

Comment:
Slices in AV1 are replaced with what they call tiles, and I think NvENC allows to control them with some limitations.
Have you looked into `enableCustomTileConfig, numTileColumns, numTileRows, tileWidths. tileHeights` parameters?

Reply:
Yep, for some reason I was misreading the docs and thought I had to calculate all the tile sizes manually if I specified the tile counts, but that's only true if I set `enableCustomTileConfig`. Added this.